common/src/main/java/io/druid/collections/BlockingPool.java

```diff
@@ -74,40 +74,56 @@ public int getPoolSize()
   }
 
   /**
-   * Take a resource from the pool.
+   * Take a resource from the pool, waiting up to the
+   * specified wait time if necessary for an element to become available.
    *
-   * @param timeout maximum time to wait for a resource, in milliseconds. Negative means do not use a timeout.
+   * @param timeout maximum time to wait for a resource, in milliseconds.
    *
    * @return a resource, or null if the timeout was reached
    */
   public ReferenceCountingResourceHolder<T> take(final long timeout)
```

Comment:
If you don't want to add a time unit parameter (which in my choice even if the method is always called with the same time unit argument, for readability on the caller side), please rename the parameter to "timeoutMs".